src/components/widgets/TimeRangeFilter/index.js

```diff
@@ -0,0 +1,156 @@
+/** @jsx hJSX */
+import {hJSX} from '@cycle/dom';
+import styles from './styles.scss';
+import moment from 'moment';
+import _ from 'lodash';
+import {Rx} from '@cycle/core';
+import {ControlledInputHook} from 'power-ui/hooks';
+
+function intent(DOM) {
+  const rangeSlider1$ = DOM
+    .select('.TimeRangeFilter input:nth-child(1)')
+    .events('input')
+    .map(ev => parseInt(ev.target.value || '0'))
+    .filter(val => !isNaN(parseInt(val)));
+
+  const rangeSlider2$ = DOM
+    .select('.TimeRangeFilter input:nth-child(2)')
+    .events('input')
+    .map(ev => parseInt(ev.target.value || '0'))
+    .filter(val => !isNaN(parseInt(val)));
+
+  return {rangeSlider1$, rangeSlider2$};
+}
+
+function inferIndex(props, rangeTime) {
+  return rangeTime.diff(props.availableTimeRange.start, 'months');
+}
+
+function makeUpdate(actions, props$) {
+  return props$.flatMap(props => {
+    const startIdx = inferIndex(props, props.selectedTimeRange.start);
+    const endIdx = inferIndex(props, props.selectedTimeRange.end);
+
+    return Rx.Observable.combineLatest(
+      actions.rangeSlider1$.startWith(startIdx),
+      actions.rangeSlider2$.startWith(endIdx),
+      (value1, value2) => {
+        return {min: Math.min(value1,value2), max: Math.max(value1,value2)};
+      })
+      .map(selection => function updateTimeRange(oldState) {
+        const selectedTimeRange = {
+          start: moment().startOf('month').add(selection.min, 'months'),
+          end: moment().startOf('month').add(selection.max, 'months').endOf('month'),
+        };
+        return {...oldState, selectedTimeRange};
+      });
```

Comment:
You can safely move this `.map(selection =>` to **outside**/after the `.flatMap(props => {  ... })`